client/allocrunner/taskrunner/task_runner.go

```diff
@@ -694,13 +694,14 @@ func (tr *TaskRunner) persistLocalState() error {
 func (tr *TaskRunner) buildTaskConfig() *drivers.TaskConfig {
 	task := tr.Task()
 	alloc := tr.Alloc()
+	taskResources:= tr.taskResources
 
 	return &drivers.TaskConfig{
 		ID:      fmt.Sprintf("%s/%s/%d", alloc.ID, task.Name, tr.restartTracker.GetCount()),
 		Name:    task.Name,
 		JobName: alloc.Job.Name,
 		Resources: &drivers.Resources{
-			NomadResources: task.Resources,
+			NomadResources: taskResources,
```

Comment:
`LinuxResources` are still being populated by the task requested resources - do we want these to also come from taskrunner?